python/ray/dag/compiled_dag_node.py

```diff
@@ -669,6 +681,12 @@ def _get_or_create_local_actor_handle():
 
         self._dag_creator_actor = _get_or_create_local_actor_handle()
 
+    def increment_max_execution_index(self) -> None:
+        """Incrment the max execution index. It is used only by
+        async reader to update _max_execution_index correctly.
```

Comment:
Looks like this is not only used by async reader, e.g., it is used in `execute_until`. Or did you mean something else when referring to "async reader"?